astropy/coordinates/representation.py

```diff
@@ -2077,6 +2182,36 @@ def from_cartesian(cls, cart):
 
         return cls(phi=phi, theta=theta, r=r, copy=False)
 
+    def transform(self, matrix):
+        """Transform the spherical coordinates using a 3x3 matrix.
+
+        This returns a new representation and does not modify the original one.
+        Any differentials attached to this representation will also be
+        transformed.
+
+        Parameters
+        ----------
+        matrix : (3,3) array-like
+            A 3x3 (or stack thereof) matrix, such as a rotation matrix.
+
+        """
+        if self.differentials:
+            # TODO! shortcut if there are differentials.
+            # Currently just super, which uses Cartesian backend.
+            rep = super().transform(matrix)
+
+        else:
+            # apply transformation in unit-spherical coordinates
+            xyz = erfa_ufunc.s2c(self.phi, 90*u.deg-self.theta)
+            p = erfa_ufunc.rxp(matrix, xyz)
+            lon, lat, ur = erfa_ufunc.p2s(p)  # `ur` is transformed unit-`r`
+
+            # create transformed physics-spherical representation,
+            # reapplying the distance scaling
+            rep = self.__class__(phi=lon, theta=90*u.deg-lat, r=self.r * ur)
```

Comment:
Does it make sense to route this through the `SphericalRepresentation.transform()`?

Reply:
It is more almost-duplicated code.